smart-contracts/contracts/airdrop/src/admin.rs

```diff
@@ -0,0 +1,365 @@
+    env: Env,
+    config: AirdropConfig,
+) -> Result<Response, AirdropErrors> {
+    // Load the current airdrop configuration from storage
+    let current_airdrop_config = AIRDROP_CONFIG.load(deps.storage)?;
+
+    // Check if an airdrop has been executed on the contract and if the update is allowed
+    if current_airdrop_config.end_height != 0
+        && env.block.height > current_airdrop_config.end_height
+    {
+        return Err(AirdropErrors::Std(StdError::GenericErr {
+            msg: "Failed to execute update as it is post airdrop ending".to_string(),
+        }));
+    }
+
+    // Check various conditions to validate the airdrop configuration update
+    validate_update_config(config.clone(), deps.storage, deps.querier, env)?;
+
+    // Save the new airdrop configuration to storage
+    AIRDROP_CONFIG.save(deps.storage, &config)?;
+
+    // Build attributes for the event
+    let attributes: Vec<Attribute> = vec![
+        Attribute {
+            key: "description".to_string(),
+            value: config.airdrop_description.to_string(),
+        },
+        Attribute {
+            key: "airdrop_amount".to_string(),
+            value: config.airdrop_amount.to_string(),
+        },
+        Attribute {
+            key: "airdrop_asset".to_string(),
+            value: config.airdrop_asset.to_string(),
+        },
+        Attribute {
+            key: "claimed".to_string(),
+            value: config.total_claimed.to_string(),
+        },
+        Attribute {
+            key: "start_height".to_string(),
+            value: config.start_height.to_string(),
+        },
+        Attribute {
+            key: "end_height".to_string(),
+            value: config.end_height.to_string(),
+        },
+    ];
+
+    // Return a default response to indicate success with an "update_airdrop_config" event
+    Ok(Response::default()
+        .add_event(Event::new("update_airdrop_config").add_attributes(attributes)))
+}
+
+/// Adds new users and their respective amounts to the airdrop configuration.
+///
+/// # Arguments
+///
+/// * `deps` - Dependencies to access storage and external data.
+/// * `users` - A vector of user addresses to be added.
+/// * `amounts` - A vector of amounts to be allocated to each user.
+///
+/// # Errors
+///
+/// Returns an error if the airdrop window is not open, the number of users and amounts provided do not match,
+/// or if any of the provided users already have existing claims or allocations.
+///
+/// # Returns
+///
+/// Returns a response indicating the success of the addition operation and includes relevant attributes
+/// in the event.
+pub fn execute_add_users(
+    deps: DepsMut,
+    users: Vec<String>,
+    amounts: Vec<Uint128>,
+) -> Result<Response, AirdropErrors> {
+    // Load the current airdrop configuration from storage
+    let current_airdrop_config = AIRDROP_CONFIG.load(deps.storage)?;
+
+    // Check if the current airdrop window is not open (start_height or end_height not zero)
```

Comment:
Add the checks if airdrop is still active.

Reply:
This can only be done if both end and start height are non zero. Which means that the airdrop has been activated once. So no need to checks if the airdrop is still active.